dynomite/src/lib.rs

```diff
@@ -274,39 +308,94 @@ pub trait Attribute: Sized {
     fn from_attr(value: AttributeValue) -> Result<Self, AttributeError>;
 }
 
+impl Attribute for AttributeValue {
```

Comment:
There are a lot of independently helpful changes in here. They are all very much appreciated. Collectively they make it a bit harder to review in a single pull.

I'm trying to separate refactorings from helpful additions from the changes that were needed for the original goal.

I'm the future let's try to break out those independent kinds of changes into independent pulls so I can help you move more quickly through iterations